### DISCUSSION TITLE
Xournal++ 1.2.4

### DISCUSSION THREAD

**Discussion by rolandlo** [xournalpp/xournalpp#5970] [General]
This is a new minor version of Xournal++ with bug fixes and improvements from the community.

  * Fixed various dark/light theme inconsistencies
  * Improved edge panning for large selections
  * Fixed various crashes, freezes and memory leaks
  * MacOS: Added support for opening xopp-files from Finder
  * Added builds for Ubuntu 24.04 and MacOS ARM

See the [changelog](https://github.com/xournalpp/xournalpp/blob/release-1.2/CHANGELOG.md) for a detailed list of all changes included.

MacOS ARM users: After installing Xournal++ in the applications folder, run the following command to remove quarantine status:
                                   xattr -d com.apple.quarantine /Applications/Xournal++.app

Windows users of the non-portable version: There are issues with the translation of some strings into languages other than English. You can get improved translations by downloading the asset locale_folder_for_non_portable_windows_version.zip and merging its content into the Xournal++\share\locale folder inside the Xournal++ installation folder.

## Changes:

* 2a4ed3fde72eb4ee66ece77ad1d46f49f26f250f Release 1.2.4
* 62f6a38af4835767aed851a092a5fc3e4f6476cf New Crowdin updates (#5957)
* cd85f54058d204eaa04653585fd3f0f7a892a272 Update changelog for 1.2.4
* c5aa08dfbc7536069a0030b84b597d431c0f6768 Unify Shape recognizer strings
* f89367c7e0a377d2d96cbdde6a0362dd2074ff26 Fix poppler includes
* 58e4cf1b71f6820cedb69fde1b0763da9a2aea22 Add testing to build script
* eaac7b8a4acf5d0135204eb3ab074685aef37318 Make task naming more explicit
* 709d4070c1423e7dfece5388460b66cfaed5af19 Add upload task
* fd7f001218a4a56268baa0a621ed92707ac9c5d3 Fix pkg-config not found
* 2b40e19d71af006640e902a7442892465f98aaea Fixups to Technius scripts
<details><summary><b>See More</b></summary>

* 9289418d8813ed3732dff0ee25ac587dff812ca0 Add Technius cirrus jhbuild script and .yml
* 1f3365a2547f8d8c7134a0eed99725ba5647651c New Crowdin updates (#5953)
* 4cd3118db656104e8ee93814ea461588ce599ba9 Fix page transition when selection is moved to another page
* 24fd391b3d45a56a76a771f620db26b770e4a51f Fix warning PageLayerPosEntry template parameter
* 3ebf00b9aa09f31dcff0ca45e3aa8dd3e75bec12 fix debug in UndoRedo
* a70fbce92a43bed5d67d409b2b621ed057e5a50a Improve filesystem error message.
* f10fb4c6862f4fce97ac50b010fb64220dc97905 Make MacOS Finder open xopp-files
* b51995e2383626653257dffb824145f0fffe144a Fix #2943 - If the presentation mode zoom value is above the maximal value, we entered an infinite loop
* d251768f5f1a30c0937709d40f9b35a08e9b2e3c Add Ubuntu 24.04 to builds
* 955b1fc880e075b8cbca77cd5825e8ae398b0b13 Fix RecentManager::getMostRecent() in case no xopp file had previously been opened (recently enough)
* e8b214c8d4229b5e6ae081b12653cc2d82b2eb2a Fix 32 bits compilation
* d55d0729f36adf75613666adf1f547748c8cff1c Fix UB with GTK signals:
* 992fb85c28b09e5c05529f4e1b5038295016f937 add uim-data to AppImage.
* 747c6bca687a18eee7cb99ae6f7f8338aae9f1c5 Bump MacOS pipeline image
* 42e40f071b4cca0ad5939ec7f4596dca43e7ef25 Allow missing terminating newlines and empty lines in .gpl files
* c26b5713b39e4d21dae98bb3a4ea56bc0faa0499 Add a check in PageView for the buffer mask [ #5659 ]
* 87cade840c214816bd8bdca91c3de2bfbb5e8c10 Add test for non-western characters
* f95ec072b12d9d9d8ad8274de217ddd2b1cadcfc Ensure that filenames in `xopp` file are encoded with UTF-8
* 4bee81fb9d35d4e55ddce276ff15cd5e68158dbb Fix `createSaveFilename` encoding
* b19ae5d8f44562d4f388ddf8cba7c12e90c7ec17 Use `wbindtextdomain` on Windows
* 74ff437e7cfbcd3b944863ad003b5962036938fe Fix MacOS titlebar color
* 5b8b85c0507ec4fa1c68c2ef9e2abb5b10eff378 Fix #5606
* 8dbb0cf81a0aecf2e5c1c47537f2654f62b72e94 Change "Stracktrace" to "Stacktrace"
* 8f50cacd1cdb26cfc2b36c1a5955f5fc56bd4cce Check for failure of popen() in stacktrace
* ad13d6aea7d0883826622b13964b06d866f6606a g_message -> g_debug for Gtk theme and locale dir
* d43b39068f4666c0f10df760f2f9c32d81a9d349 Fix theme variant on MacOS when GTK_THEME is set
* 1ef8b59bc0fd3a3bf00de8d66354c1cb422c5562 Fix various Dark/Light Theme inconsistencies (#5583)
* 06495644b6ecc894e77e1248ec5298f80b85ef8b Fixed Memory leaks for some GLibGuards, added out_ptr
* 33fee4ce287dd9af4774bf107e89804b0ad8283e Smoother edge panning on large selections (#5578)
* 0543aa42ec740681eb7f73a8562e060227fa983e Fix path separator for Windows
* 45dafa963888ff7349683d14f748767a92817f38 Make "Replace File?" dialog transient for its parent dialog
* f5bb01d00f19f3f41a5b24f913787c181c0a2f6d Make ok button default in export dialog
* 83f05d37c9316162f64af49d888053e3444c36e3 Bump the preview buffer size to worst case scenario. [ #5534 ]
* cb6bd981d82bfb3ab3de0e2bcf80fa2f69582583 Automated version bump to 1.2.4~dev

This list of changes was [auto generated](https://dev.azure.com/xournalpp/ce85de88-4b71-44a0-8a57-2707c0b9a6e7/_release?releaseId=854&_a=release-summary).</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/xournalpp/xournalpp/releases/tag/v1.2.4'>Xournal++ 1.2.4</a>.</em>

**Comment by joaodinizgit** [comment]
Thank you very much! I downloaded this version and I'm really like it.😀
